conans/client/cmd/export_pkg.py

```diff
@@ -43,7 +44,12 @@ def export_pkg(cache, graph_manager, hook_manager, recorder, output,
         packager.export_pkg(conanfile, package_id, package_folder, dest_package_folder,
                             hook_manager, conan_file_path, ref)
     else:
-        packager.create_package(conanfile, package_id, source_folder, build_folder,
-                                dest_package_folder, install_folder, hook_manager, conan_file_path,
-                                ref, local=True)
+        with cache.package_layout(ref).update_metadata() as metadata:
+            packager.create_package(conanfile, package_id, source_folder, build_folder,
+                                    dest_package_folder, install_folder, hook_manager, conan_file_path,
+                                    ref, local=True)
```

Comment:
This should be outside the `with ... metadata:` block, two reasons: 1) it is not used, and 2) there are _hooks_ been called from inside that function and we are considering passing the metadata to hooks, we would lose any changes/additions a hook could do to the `metadata` 👀